Library/Homebrew/keg_relocate.rb

```diff
@@ -174,16 +162,6 @@ def find_dylib(bad_name)
     lib.find { |pn| break pn if pn.to_s.end_with?(suffix) }
   end
 
-  def mach_o_files
-    mach_o_files = []
-    path.find do |pn|
-      next if pn.symlink? || pn.directory?
-      mach_o_files << pn if pn.dylib? || pn.mach_o_bundle? || pn.mach_o_executable?
-    end
-
-    mach_o_files
-  end
```

Comment:
> This could be fixed by guarding the usage of methods like Pathname#dylib? with a file sanity check

Let's do this. It's strange to let a non dylib file raise exception instead of return false when calling `dylib?`.